server/include/bge/Component.h

```diff
@@ -33,16 +37,53 @@ namespace bge {
 
     struct MovementRequestComponent : Component<MovementRequestComponent> {
         MovementRequestComponent(bool forwardRequested, bool backwardRequested, bool leftRequested, bool rightRequested, bool jumpRequested, float pitch, float yaw)
-            : forwardRequested(forwardRequested), backwardRequested(backwardRequested), leftRequested(leftRequested), rightRequested(rightRequested), jumpRequested(jumpRequested), pitch(pitch), yaw(yaw) {
+            : forwardRequested(forwardRequested), backwardRequested(backwardRequested), leftRequested(leftRequested), rightRequested(rightRequested), 
+                jumpRequested(jumpRequested), pitch(pitch), yaw(yaw) {
         }
         bool forwardRequested, backwardRequested, leftRequested, rightRequested, jumpRequested;
         float yaw, pitch;
+        glm::vec3 forwardDirection;
+    };
+
+
+    struct HealthComponent : Component<HealthComponent> {
+        HealthComponent(int healthPoint) : healthPoint(healthPoint) {
+        }
+        int healthPoint;
+    };
+
+    /**
+    * each entity that we want to do box collision on must have dimension component
+    * we assume that these entity are cylinder-shaped
```

Comment:
Are we actually doing cylinder collisions? It seems like the collisions are based on rectangular prisms, but I may have misunderstood the code.

Reply:
Youre right. No cylinders. I think Anh added this comment. will delete this